sql/solutions/lesson4.md

```diff
@@ -0,0 +1,125 @@
+---
+layout: page
+title: Introduction to SQL Solutions
+---
+# Lesson 1 exercises Solutions
+
+## L4.1 Update the age of the person named Luis Marco Polo.
+
+Solution 1:
+First find person's id:
+```SQL
+SELECT *
+  FROM people
+ WHERE name = 'Luis Marco Polo';
+```
+Update age using id number:
+```SQL
+UPDATE people
+   SET age = age + 1
+ WHERE id = 3;
+```
+
+Solution 2:
+Just use their name as identifier to update age. (When could this cause problems?)
+```SQL
+UPDATE people
+   SET age = age + 1
+ WHERE name = 'Luis Marco Polo';
+```
+## L4.2 Increase by 1 the age of all the people who were born in Italy.
+
+```SQL
+UPDATE people
+   SET age = age + 1
+ WHERE country_of_birth = 'IT';
+```
+
+## L4.3 Set all names to lowercase.
+
+```SQL
+UPDATE people
+   SET name = LOWER(name);
+```
+
+## L4.4 Insert a new student.
+
+Solution 1:
+What problems could arise with this solution?
+```SQL
+INSERT INTO people (id, name)
+VALUES (42, 'Valerie Knight');
+```
+
+Solution 2:
+```SQL
+INSERT INTO people (id, name)
+VALUES (
+	   (SELECT id
+          FROM people
+         ORDER BY id DESC
+         LIMIT 1) + 1,
+       'Valerie Knight');
+```
+
+
+## L4.5 Delete the record you just inserted.
+
+Solution 1:
+What problems could arise with this solution?
+```SQL
+DELETE FROM people
+WHERE id = 45;
+```
+
+Solution 2:
+```SQL
+DELETE FROM people
+ WHERE id =
+	   (SELECT id
+          FROM people
+         ORDER BY id DESC
+         LIMIT 1);
```

Comment:
Solution 2 works, but is very risky to use in real life, you are deleting something not explicitly. What would happen for example if someone else adds another person before you run this query? you'll be removing the other person record instead of yours